src/lib/engine/index.ts

```diff
@@ -61,6 +66,69 @@ export function runGame(code: string, canvas: HTMLCanvasElement, onPageError: (e
 			const playTuneRes = playTune(tune, n)
 			tunes.push(playTuneRes)
 			return playTuneRes
+		},
+		console: {
+			...console,
+			log: (...args: any[]) => {
+				console.log(...args)
+				logInfo.value = [...logInfo.value, {
+					args: args,
+					nums: (()=>{
+						var err = getErrorObject();
+						// get <anonymous>:x:x if on chrome, Function:x:x on firefox
+						// @ts-ignore
+						var stack = err.stack
+						// check browser
+						var browser = navigator.userAgent;
+						if(browser.indexOf("Chrome") != -1) {
+							// chrome
+							var offset = stack.indexOf("<anonymous>:") + 12;
+							// slice from there until ')'
+							var line = stack.slice(offset, stack.indexOf(")", offset))
+							var strs = line.split(":")
+							return [parseInt(strs[0]) - 2, parseInt(strs[1])]
+						} else {
+							// firefox/etc.
+							var offset = stack.indexOf("Function:") + 9;
```

Comment:
This seems brittle...

Reply:
funnily enough it's actually better than the sprig editor's way of doing it (unfortunately), it is also the only physical possible way this can be done (the JS runtime does not expose line numbers/col ANYWHERE else)